### PR TITLE
RUM-3458: Fix time drift in RecordedDataQueueHandler

### PR DESCRIPTION
**Pull request by jonathanmos**
### What does this PR do?
Fixes an issue in the RecordedDataQueueHandler where the "current" time was being passed in to the loop, but never updated inside it. This could lead to the time drifting when the queue was very long, leading to items being considered out of date.

### Motivation
We have telemetry indicating numerous DataQueueItems are being dropped for being out of date. 

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/async/RecordedDataQueueHandler.kt [hunk 1]
```diff
@@ -170,8 +124,14 @@ internal class RecordedDataQueueHandler : DataQueueHandler {
...
     @Synchronized
-    private fun triggerProcessingLoop(currentTime: Long) {
+    private fun triggerProcessingLoop(initialTime: Long) {
+        var currentTime = initialTime
+
         while (recordedDataQueue.isNotEmpty()) {
+            mainThreadHandler.post {
+                currentTime = timeProvider.getDeviceTimestamp()
+            }
+
```
Comment (0xnm) [inline features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/async/RecordedDataQueueHandler.kt:134]: I'm not sure it is a right solution. Maybe it can reduce the problem somehow, but the issue here is concurrency: `mainThreadHandler.post` is non-blocking call, so by the time `currentTime = timeProvider.getDeviceTimestamp()` is executed, you can be already advance in the `while` loop past `post` line, or even be on the next turn.

Comment (jonathanmos) [reply features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/async/RecordedDataQueueHandler.kt]: I've changed the implementation to have a separate property to store `System.nanoTime` for each item and query that, so it will be monotonic and we're no longer dependent upon calling on a certain thread

File: features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/SessionReplayRecorder.kt [hunk 2]
```diff
@@ -104,8 +109,23 @@ internal class SessionReplayRecorder : OnWindowRefreshedCallback, Recorder {
         this.recordedDataQueueHandler = RecordedDataQueueHandler(
             processor = processor,
             rumContextDataHandler = rumContextDataHandler,
-            timeProvider = timeProvider,
-            internalLogger = internalLogger
+            internalLogger = internalLogger,
+
+            /**
+             * TODO RUMM-0000 consider change to LoggingThreadPoolExecutor once V2 is merged.
```
Comment (0xnm) [inline features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/SessionReplayRecorder.kt:115]: I think it is time to do it: it is a simple change and can be made in another PR

**Review by 0xnm — APPROVED** [review]
lgtm. Since we care only about item age at this point, using 2 `System.nanoTime()` instants is fine and result of this call is _monotonically_ increasing.

However, there is a second timestamp involved, which is attached to the queue item and then used later to build wireframes (`nextItem.recordedQueuedItemContext.timestamp`), so it is better to have another pair of eyes for the review here, because I'm don't have a deep knowledge of the whole processing pipeline.

File: features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/SessionReplayRecorder.kt (hunk 2 above)
Comment (xgouchet) [inline features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/SessionReplayRecorder.kt:115]: > [!CAUTION]
> Do not use a "RUM-000" ticket, actually create one and reference it here.

**Review by xgouchet — CHANGES_REQUESTED** [review]

File: features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/SessionReplayRecorder.kt (hunk 2 above)
Comment (jonathanmos) [reply features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/SessionReplayRecorder.kt]: 👍  I've opened a ticket and modified this to reference

**Review by xgouchet — APPROVED** [review]
